pb-rs/src/types.rs

```diff
@@ -2145,24 +2158,37 @@ impl FileDescriptor {
             )?;
             return Ok(());
         }
-        writeln!(w, "use std::io::Write;")?;
-        if self
-            .messages
-            .iter()
-            .any(|m| m.all_fields().any(|f| f.typ.has_cow()))
-        {
-            writeln!(w, "use std::borrow::Cow;")?;
+
+        if config.nostd {
+            writeln!(w, "extern crate alloc;")?;
```

Comment:
same comment for the repeated `extern crate` calls. Usually the files generated by pb-rs will be modules to be inserted into wider project. I believe the extern crate declarations should be in the `lib.rs` or `main.rs` file, not here. We could probably trigger some cli message or add a comment in every file?

Reply:
I think agree with this. The usability of requiring an `extern crate ...` in your code to use no_std is sub-optimal, but I like the explicitness of it. I don't really like changing global properties because you included some generated code.

Reply:
see the official take on this from when alloc was stabilised:
https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html#the-alloc-crate-is-stable
so this should be on lib.rs, yes.